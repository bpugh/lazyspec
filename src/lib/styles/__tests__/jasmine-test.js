const fs = require('fs');

const describeModule = require('../../describeModule');
const specUnit = require('../jasmine');

function expectSpec(moduleName) {
  const src = fs.readFileSync(`spec/fixtures/${moduleName}.js`).toString();
  const description = describeModule(moduleName, src);

  const spec = specUnit(description);

  expect(spec).toMatchSnapshot();
}

describe('describeModule', () => {
  it('should describe simple exports', () => {
    expectSpec('esExports');
  });

  it('should describe a createClass component', () => {
    expectSpec('CreateClass');
  });
});
