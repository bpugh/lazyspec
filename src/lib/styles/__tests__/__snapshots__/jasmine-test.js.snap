// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`describeModule should describe a createClass component 1`] = `
"
describe('CreateClass', () => {
  it('should exist', () => {
    expect(CreateClass).toBeTruthy();
  });
});
"
`;

exports[`describeModule should describe simple exports 1`] = `
"
describe('esExports', () => {
  it('should exist', () => {
    expect(esExports).toBeTruthy();
  });
});


describe('refConst', () => {
  it('should exist', () => {
    expect(refConst).toBeTruthy();
  });
});


describe('refFunc', () => {
  it('should exist', () => {
    expect(refFunc).toBeTruthy();
  });
});


describe('inlineConst', () => {
  it('should exist', () => {
    expect(inlineConst).toBeTruthy();
  });
});


describe('inlineFunc', () => {
  it('should exist', () => {
    expect(inlineFunc).toBeTruthy();
  });
});
"
`;
